oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/storage/PersistenceTimer.java

```diff
@@ -118,9 +128,17 @@ private void extractDataAndSave(IBatchDAO batchDAO) {
 
             HistogramMetrics.Timer executeLatencyTimer = executeLatency.createTimer();
             try {
-                if (CollectionUtils.isNotEmpty(prepareRequests)) {
-                    batchDAO.synchronous(prepareRequests);
+                List<List<PrepareRequest>> partitions = Lists.partition(prepareRequests, maxSyncoperationNum);
+                CountDownLatch countDownLatch = new CountDownLatch(partitions.size());
+                for (final List<PrepareRequest> partition : partitions) {
+                    executorService.submit(() -> {
+                        if (CollectionUtils.isNotEmpty(partition)) {
+                            batchDAO.synchronous(partition);
+                        }
```

Comment:
Let's enclose this inside a `try-catch` block, in case the `latch` is never counted down when exception occurs.